cub/test/catch2_test_device_reduce.cu

```diff
@@ -0,0 +1,464 @@
+
+    // Run test
+    using result_t = cub::KeyValuePair<int, output_t>;
+    thrust::device_vector<result_t> out_result(num_segments);
+    device_arg_max(d_in_it, thrust::raw_pointer_cast(out_result.data()), num_items);
+
+    // Verify result
+    result_t gpu_result = out_result[0];
+    REQUIRE(expected_result[0] == gpu_result.value);
+    REQUIRE((expected_result - host_items.cbegin()) == gpu_result.key);
+  }
+
+  SECTION("argmin")
+  {
+    // Prepare verification data
+    thrust::host_vector<item_t> host_items(in_items);
+    auto expected_result = std::min_element(host_items.cbegin(), host_items.cend());
+
+    // Run test
+    using result_t = cub::KeyValuePair<int, output_t>;
+    thrust::device_vector<result_t> out_result(num_segments);
+    device_arg_min(d_in_it, thrust::raw_pointer_cast(out_result.data()), num_items);
+
+    // Verify result
+    result_t gpu_result = out_result[0];
+    REQUIRE(expected_result[0] == gpu_result.value);
+    REQUIRE((expected_result - host_items.cbegin()) == gpu_result.key);
+  }
+}
+
+// TEST_TYPES == 4
+#elif TEST_TYPES == 4
+
+CUB_TEST("Device sum works for big indices", "[reduce][device]")
+{
+  test_big_indices_helper<std::size_t, std::uint32_t>(1ull << 30);
+  test_big_indices_helper<std::size_t, std::uint32_t>(1ull << 31);
+  test_big_indices_helper<std::size_t, std::uint32_t>((1ull << 32) - 1);
+  test_big_indices_helper<std::size_t, std::uint64_t>(1ull << 33);
+}
+
+CUB_TEST("Device reduce arg{min,max} works with inf items", "[reduce][device]")
+{
+  using in_t     = float;
+  using offset_t = int;
+  using out_t    = cub::KeyValuePair<offset_t, in_t>;
+
+  const int n     = 10;
+  const float inf = ::cuda::std::numeric_limits<float>::infinity();
+
+  thrust::device_vector<out_t> out(1);
+  out_t *d_out = thrust::raw_pointer_cast(out.data());
+
+  /**
+   * ArgMin should return max value for empty input. This interferes with
+   * input data containing infinity values. This test checks that ArgMin
+   * works correctly with infinity values.
+   */
+  SECTION("InfInArgMin")
+  {
+    const int n     = 10;
+    const float inf = ::cuda::std::numeric_limits<float>::infinity();
+
+    thrust::device_vector<in_t> in(n, inf);
+    const in_t *d_in = thrust::raw_pointer_cast(in.data());
+
+    device_arg_min(d_in, d_out, n);
+
+    const out_t result = out[0];
+    REQUIRE(result.key == 0);
+    REQUIRE(result.value == inf);
+  }
+
+  /**
+   * ArgMax should return lowest value for empty input. This interferes with
+   * input data containing infinity values. This test checks that ArgMax
+   * works correctly with infinity values.
+   */
+  SECTION("InfInArgMax")
+  {
+    thrust::device_vector<in_t> in(n, -inf);
+    const in_t *d_in = thrust::raw_pointer_cast(in.data());
+
+    device_arg_max(d_in, d_out, n);
+
+    const out_t result = out[0];
+    REQUIRE(result.key == 0);
+    REQUIRE(result.value == -inf);
+  }
+}
+
+// TEST_TYPES == 5
+#elif TEST_TYPES == 5
```

Comment:
This approach is not very development-friendly. If you need different tests in different TUs, can we consider putting them into different source files?

Reply:
Agreed! Thanks for the suggestion. Moved the two test sections into their own translation unit:
- `device_reduce_iteraturs.cu`: 64-bit indices and fancy iterators
- `device_reduce_fp_inf.cu`: `Arg{Min,Max}` with `inf`
Would be great if I could have your eyes do another pass over this refactor quickly 🙏